### PR TITLE
NAS-121916 / 23.10 / Fix displaying system serial for HA systems

### PR DESCRIPTION
**Pull request by denysbutenko**
For more context please check original ticket.

For testing, use the M50 system from a ticket. First, ensure the `System Information` and `System Information(standby)` widgets show the correct system serial number. Then, go to `System -> General` and check the `Support` card if the `System Serial` has two different serials of HA nodes. Initiate failover and recheck it.

### CODE REVIEW

**Review by dszidi — CHANGES_REQUESTED** [review]
Code looks good and both sysinfo widget and support cards are showing the correct values.

After testing it on our team's M40 I noticed a situation that might cause confusion for users. The value shown on the dashboard is basically the hardware serial and the support card shows the serials the license was issued for. Under normal situations these two will match however, there are possible cases where they might not (RMA, Dev License, Human error etc).

To avoid confusion, please change the label on the support card from "System Serial" to "Licensed Serials"

**Comment by denysbutenko** [comment]
> Code looks good and both sysinfo widget and support cards are showing the correct values.
> 
> After testing it on our team's M40 I noticed a situation that might cause confusion for users. The value shown on the dashboard is basically the hardware serial and the support card shows the serials the license was issued for. Under normal situations these two will match however, there are possible cases where they might not (RMA, Dev License, Human error etc).
> 
> To avoid confusion, please change the label on the support card from "System Serial" to "Licensed Serials"

Ok. I assume that if the system has a license (non-HA) it should display the license serial as well. Otherwise, it will show `Serial Number`. Is that correct?

**Comment by denysbutenko** [comment]
Adapted the `system.is_ix_hardware` endpoint and added `Licensed Serials` along with `System Serial`.

**Comment by denysbutenko** [comment]
@dszidi, please take another look at it.

File: src/app/pages/system/general-settings/support/support-card/support-card.component.ts [hunk 1]
```diff
@@ -110,7 +110,9 @@ export class SupportCardComponent implements OnInit {
     const expDateConverted = new Date(this.licenseInfo.contract_end.$value);
     this.licenseInfo.expiration_date = this.licenseInfo.contract_end.$value;
     if (this.licenseInfo?.system_serial_ha) {
-      this.systemInfo.serial = this.systemInfo.system_serial + ' / ' + this.licenseInfo.system_serial;
+      this.systemInfo.serial = this.licenseInfo.system_serial + ' / ' + this.licenseInfo.system_serial_ha;
+    } else if (this.licenseInfo?.system_serial) {
+      this.systemInfo.serial = this.licenseInfo.system_serial;
     } else {
       this.systemInfo.serial = this.systemInfo.system_serial;
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/pages/system/general-settings/support/support-card/support-card.component.ts]: This condition was removed. The template gets serials directly from the data. If there is `this.licenseInfo.system_serial` or `this.licenseInfo.system_serial_ha` are null, then the `Licensed Serials` row won't be rendered on the card. The hardware serial (`System Serial`) has its row and no longer depends on the license.

File: src/app/pages/dashboard/components/widget-sys-info/widget-sys-info.component.ts [hunk 2]
```diff
@@ -234,7 +230,7 @@ export class WidgetSysInfoComponent extends WidgetComponent implements OnInit {
   }
 
   setProductImage(data: SystemInfo): void {
-    if (this.manufacturer !== 'ixsystems') return;
+    if (!this.isIxHardware) return;
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/pages/dashboard/components/widget-sys-info/widget-sys-info.component.ts:234]: Fixed

**Review by dszidi — APPROVED** [review]
Looks good. Now these should all make sense to the user

image


image



image

**Pull request merged**